lib/core/src/chain/bitcoin.rs

```diff
@@ -114,30 +114,28 @@ impl BitcoinChainService for HybridBitcoinChainService {
                 // It might be that the client has reconnected and subscriptions don't persist
                 // across connections. Calling `client.ping()` won't help here because the
                 // successful retry will prevent us knowing about the reconnect.
-                if let Ok(header) = self.client.block_headers_subscribe_raw() {
+                if let Ok(header) = client.block_headers_subscribe_raw() {
                     Some(header.try_into()?)
                 } else {
                     None
                 }
             }
         };
 
-        let mut tip = self.tip.lock().unwrap();
-        if let Some(new_tip) = new_tip {
-            *tip = new_tip;
-        }
-
-        Ok(tip.clone())
+        let tip = new_tip.ok_or_else(|| anyhow!("Failed to get tip"))?;
```

Comment:
Nit: the `new_tip` name no longer makes sense. Maybe rename to `maybe_tip`? We can even get rid of it and return the tip directly within the `match`.